nilearn/surface/surface.py

```diff
@@ -157,26 +172,33 @@ def _ball_sample_locations(
 
     """
     if depth is not None:
-        raise ValueError("The 'ball' sampling strategy does not support "
-                         "the 'depth' parameter")
+        raise ValueError(
+            "The 'ball' sampling strategy does not support "
+            "the 'depth' parameter"
```

Comment:
The error message is not great. Maybe we could take this opportunity to improve it...

Reply:
will try to improve it

Reply:
I at least added a tip on how to avoid the error.

IMO this error should just be a warning.